Software/Program/Code/Core/Src/main.c

```diff
@@ -0,0 +1,589 @@
+/* USER CODE BEGIN Header */
+/**
+  ******************************************************************************
+  * @file           : main.c
+  * @brief          : Main program body
+  ******************************************************************************
+  * @attention
+  *
+  * Copyright (c) 2024 STMicroelectronics.
+  * All rights reserved.
+  *
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+#define PUSH_BUTTON_DEBOUNCE_WAIT_TIME              50  // Amount of time _after_ an edge (on the digital input connected to the push button) to ignore any level changes
+// Value to load into the ARR (Auto Reload Register) at startup:
+#define AUTO_RELOD_REGISTER_INITIAL_VALUE           ( (uint32_t) ((DIRECTION_FORWARD_HIGH_PULSE_TIME_NOMINAL - DIRECTION_FORWARD_HIGH_PULSE_TIME_TOLERANCE) * \
+                                                                   TIMER_COUNTS_PER_NANOSECOND) )
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+ADC_HandleTypeDef hadc3;
+
+TIM_HandleTypeDef htim2;
```

Comment:
```suggestion
static ADC_HandleTypeDef hadc3;

static TIM_HandleTypeDef htim2;
```
Should surely be file-scope, possibly also `const`?